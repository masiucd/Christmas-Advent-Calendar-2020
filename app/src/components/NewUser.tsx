import * as React from 'react';
import UserForm from './UserForm';

export interface INewUserProps {}

export default function NewUser(props: INewUserProps) {
  return (
    <div data-testid='newUser'>
      <h1 data-testid='page-title'>New User</h1>
      <UserForm />
    </div>
  );
}
