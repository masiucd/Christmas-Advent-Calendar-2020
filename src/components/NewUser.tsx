import * as React from 'react';
import UserForm from './UserForm';

export interface INewUserProps {}

export default function NewUser(props: INewUserProps) {
  const handleSubmit = (e: React.FormEvent<HTMLFormElement>, text: string) => {
    e.preventDefault();
    console.log(text);
  };
  return (
    <div data-testid='newUser'>
      <h1 data-testid='page-title'>New User</h1>
      <UserForm onHandleSubmit={handleSubmit} />
    </div>
  );
}
