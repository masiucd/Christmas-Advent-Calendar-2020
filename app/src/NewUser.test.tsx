import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import NewUser from './components/NewUser';

//  clean the DOM after each test!
afterEach(cleanup);

describe('testing newUser component', () => {
  test('<NewUser/>', () => {
    const { debug, container, getByTestId, queryByTestId } = render(
      <NewUser />,
    );

    const pageTitle = getByTestId('page-title');
    const userForm = queryByTestId('user-form');

    expect(pageTitle.tagName).toBe('H1');
    expect(pageTitle.textContent).toBe('New User');
    expect(userForm).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
    // debug();
  });
});
