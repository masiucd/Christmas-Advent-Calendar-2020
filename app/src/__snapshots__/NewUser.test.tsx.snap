// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing newUser component <NewUser/> 1`] = `
<div
  data-testid="newUser"
>
  <h1
    data-testid="page-title"
  >
    New User
  </h1>
  <form
    data-testid="user-form"
  >
    <div
      class="form-grop"
    >
      <input
        type="text"
      />
    </div>
    <div
      class="form-grop"
    >
      <input
        type="text"
      />
    </div>
    <div
      class="form-grop"
    >
      <input
        type="text"
      />
    </div>
    <button
      type="submit"
    />
  </form>
</div>
`;
